src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -2373,6 +2373,52 @@ public int Property
                 verify: ExecutionConditionUtil.IsCoreClr ? Verification.Passes : Verification.Fails);
         }
 
+        [Fact]
+        public void ReadonlyFields_TypesDifferingNullability()
+        {
+            string source = @"
+public class C
+{
+    public static void Main()
+    {
+        System.Console.Write(C1<int>.F1.content);
+        System.Console.Write("" "");
+        System.Console.Write(C2<int>.F1.content);
+    }
+}
+
+public struct Container
+{
+    public int content;
+}
+
+class C1<T>
+{
+    public static readonly Container F1;
+
+    static C1()
+    {
+        C1<T>.F1.content = 2;
+    }
+}
+
+#nullable enable
+
+class C2<T>
+{
+    public static readonly Container F1;
+
+    static C2()
+    {
+        C2<T>.F1.content = 3;
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "2 3", verify: Verification.Skipped /* init-only */);
```

Comment:
>verify: Verification.Skipped /* init-only */ [](start = 58, length = 44)

This isn't an init-only scenario, no need to skip verification, I think.

Reply:
I think you're right. I did run into a PE verification error though.

As far as I can tell we're emitting the same IL [already](https://sharplab.io/#v2:C4LglgNgPgAgzAAhgJgQYQLACgDe2EFLLL6F5aGVICMAdGgPYB2AzgxAKa0DqATmMA4AKNNQA8YJsAB8tAGJ0Axs0FSAlAG5SVGvWZtOPfoKEAiBKc3aqMOo1bsufAcLTIJU2QtrKpHdVoUhAC+2KFY2PAILMC8AK6KwOgqAIaSHLzY5JRRkkm+qsCB4ZGoomIAKtJZ1lG2AGwIvBwpACbMEACeyVJpTBkICoG11I2iQmrW2Tro4lXySir+SQC8CMiBlOElWADETHEQECkARpwI/qec2KXo7lU1QQR1o00t7UxdPcB9A0M3TxoY2QEym1kobkqXkWfikCDWcE2ITC2CAA===) and in non-generic scenario [too](https://sharplab.io/#v2:C4LglgNgPgAgzAAhgJgQYQLACgDe2EFKIwCMAbAgE4CmAhgCYD2AdhAJ7ovC1jPWUIAYiQDc+QqQpoAFAEpxBPFkIr0AOmFqAxl2rNgCALwJkY5YQC+ChNZTJrS1RJJq0LAM6MI1NQHVKYMDU0mgaLjr6esCyZipWWPHY8AjuwJQArloGbvo8fJTYjhKIvAYRQfpm8UA) (which also fails PE verification). 

Feels like a PE verification bug. Is there a set of IL verification rules that I could reference to confirm this?

```
            // [ : C::Main][mdToken=0x6000004][offset 0x00000001] Cannot change initonly field outside its .ctor.
            v.VerifyIL("C.Main", @"
{
  // Code size       45 (0x2d)
  .maxstack  1
  IL_0000:  nop
  IL_0001:  ldsflda    ""Container C1<int>.F1""
  IL_0006:  ldfld      ""int Container.content""
  IL_000b:  call       ""void System.Console.Write(int)""
  IL_0010:  nop
  IL_0011:  ldstr      "" ""
  IL_0016:  call       ""void System.Console.Write(string)""
  IL_001b:  nop
  IL_001c:  ldsflda    ""Container C2<int>.F1""
  IL_0021:  ldfld      ""int Container.content""
  IL_0026:  call       ""void System.Console.Write(int)""
  IL_002b:  nop
  IL_002c:  ret
}
");
 ```